### ISSUE TITLE
Resolve issue with dependencies that prevent github workflow from completing

### ISSUE THREAD

**Issue by entrotech** [hackforla/tdm-calculator#614]
### Overview
The github workflow is failing with the messages shown below.

### Action Items
Reolve

### Resources/Instructions
```
Step 1/20 : FROM node:alpine as clientBuilder
alpine: Pulling from library/node
cbdbe7a5bc2a: Already exists
c36334c36d30: Pulling fs layer
dc731be5fee1: Pulling fs layer
8a23f73dde1a: Pulling fs layer
8a23f73dde1a: Verifying Checksum
8a23f73dde1a: Download complete
dc731be5fee1: Verifying Checksum
dc731be5fee1: Download complete
c36334c36d30: Verifying Checksum
c36334c36d30: Download complete
c36334c36d30: Pull complete
dc731be5fee1: Pull complete
8a23f73dde1a: Pull complete
Digest: sha256:90dfaf5806a607e81cc81b42b474626612c9271eee7a4fc6644ce6299348425a
Status: Downloaded newer image for node:alpine
 ---> 1e8b781248bb
Step 2/20 : RUN mkdir /app
 ---> Running in d7378f4b76f2
Removing intermediate container d7378f4b76f2
 ---> f44aec2c3666
Step 3/20 : WORKDIR /app
 ---> Running in 27bfc2527c91
Removing intermediate container 27bfc2527c91
 ---> 6750b3db4bb0
Step 4/20 : COPY /client/package.json .
 ---> 558538bb21f8
Step 5/20 : RUN npm install
 ---> Running in 038f97173c11
npm notice 
npm notice New patch version of npm available! 7.0.3 -> 7.0.5
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.0.5>
npm notice Run `npm install -g npm@7.0.5` to update!
npm notice 
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: typescript@4.0.3
npm ERR! node_modules/typescript
npm ERR!   typescript@"^4.0.3" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional typescript@"^3.2.1" from react-scripts@3.4.4
npm ERR! node_modules/react-scripts
npm ERR!   react-scripts@"^3.4.3" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /root/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-10-26T16_52_06_325Z-debug.log
The command '/bin/sh -c npm install' returned a non-zero code: 1
Error: exit status 1
Usage:
  github-actions build-push [flags]

Flags:
  -h, --help   help for build-push

exit status 1
```

**Comment by entrotech** [comment]
First attempt:
npm uninstall react-modal from /package.json

upgrade to react-scripts@4.0.0
resolve npm audit vulnerabilities.

**Comment by entrotech** [comment]
I made several attempts to resolve this, but was not successful. At first, the errors reported in the Github action log were related to ambiguous versions of jest and other npm packages. I moved all the server code to a /server folder, in hopes of resolving some of these conflicts as well as improving the project organization.  Now the remaining problem in the log is more obscure, and appears to revolve around gyp, the compiler node uses to build some npm packages like bcrypt under some circumstances.

```
1m 28s
Warning: Input 'repository' has been deprecated with message: v2 is now available through docker/build-push-action@v2
Run docker/build-push-action@v1
/usr/bin/docker run --name dockergithubactionsv1_c693a2 --label 1e5c35 --workdir /github/workspace --rm -e INPUT_USERNAME -e INPUT_PASSWORD -e INPUT_REPOSITORY -e INPUT_DOCKERFILE -e INPUT_TAG_WITH_REF -e INPUT_TAG_WITH_SHA -e INPUT_ADD_GIT_LABELS -e INPUT_LABELS -e INPUT_REGISTRY -e INPUT_TAGS -e INPUT_PATH -e INPUT_TARGET -e INPUT_ALWAYS_PULL -e INPUT_BUILD_ARGS -e INPUT_CACHE_FROMS -e INPUT_PUSH -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_PATH -e GITHUB_ENV -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/tdm-calculator/tdm-calculator":"/github/workspace" docker/github-actions:v1  "build-push"
Logging in to registry 
WARNING! Using --password via the CLI is insecure. Use --password-stdin.
WARNING! Your password will be stored unencrypted in /github/home/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

Login Succeeded
Building image [tdmcalc/tdmapp:experimental tdmcalc/tdmapp:sha-11d8e79]
Sending build context to Docker daemon  35.99MB

Step 1/26 : FROM node:alpine as clientBuilder
alpine: Pulling from library/node
cbdbe7a5bc2a: Already exists
c36334c36d30: Pulling fs layer
dc731be5fee1: Pulling fs layer
8a23f73dde1a: Pulling fs layer
8a23f73dde1a: Verifying Checksum
8a23f73dde1a: Download complete
dc731be5fee1: Verifying Checksum
dc731be5fee1: Download complete
c36334c36d30: Verifying Checksum
c36334c36d30: Download complete
c36334c36d30: Pull complete
dc731be5fee1: Pull complete
8a23f73dde1a: Pull complete
Digest: sha256:90dfaf5806a607e81cc81b42b474626612c9271eee7a4fc6644ce6299348425a
Status: Downloaded newer image for node:alpine
 ---> 1e8b781248bb
Step 2/26 : RUN mkdir /app
 ---> Running in e1123804b1e4
Removing intermediate container e1123804b1e4
 ---> b5d057b1f60c
Step 3/26 : WORKDIR /app
 ---> Running in 363cfb4e0e0d
Removing intermediate container 363cfb4e0e0d
 ---> 0b4a3f3bbdfd
Step 4/26 : COPY /client/package.json .
 ---> 2a733069fdbb
Step 5/26 : COPY /client/package-lock.json .
 ---> 3b31bc3cbd84
Step 6/26 : RUN npm ci
 ---> Running in dd42aa79962d
npm notice 
npm notice New patch version of npm available! 7.0.3 -> 7.0.6
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.0.6>
npm notice Run `npm install -g npm@7.0.6` to update!
npm notice 
npm ERR! code 1
npm ERR! path /app/node_modules/node-sass
npm ERR! command failed
npm ERR! command sh -c node scripts/build.js
npm ERR! Building: /usr/local/bin/node /app/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp verb cli [
npm ERR! gyp verb cli   '/usr/local/bin/node',
npm ERR! gyp verb cli   '/app/node_modules/node-gyp/bin/node-gyp.js',
npm ERR! gyp verb cli   'rebuild',
npm ERR! gyp verb cli   '--verbose',
npm ERR! gyp verb cli   '--libsass_ext=',
npm ERR! gyp verb cli   '--libsass_cflags=',
npm ERR! gyp verb cli   '--libsass_ldflags=',
npm ERR! gyp verb cli   '--libsass_library='
npm ERR! gyp verb cli ]
npm ERR! gyp info using node-gyp@3.8.0
npm ERR! gyp info using node@15.0.1 | linux | x64
npm ERR! gyp verb command rebuild []
npm ERR! gyp verb command clean []
npm ERR! gyp verb clean removing "build" directory
npm ERR! gyp verb command configure []
npm ERR! gyp verb check python checking for Python executable "python2" in the PATH
npm ERR! gyp verb `which` failed Error: not found: python2
npm ERR! gyp verb `which` failed     at getNotFoundError (/app/node_modules/which/which.js:13:12)
npm ERR! gyp verb `which` failed     at F (/app/node_modules/which/which.js:68:19)
npm ERR! gyp verb `which` failed     at E (/app/node_modules/which/which.js:80:29)
npm ERR! gyp verb `which` failed     at /app/node_modules/which/which.js:89:16
npm ERR! gyp verb `which` failed     at /app/node_modules/isexe/index.js:42:5
npm ERR! gyp verb `which` failed     at /app/node_modules/isexe/mode.js:8:5
npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:183:21)
npm ERR! gyp verb `which` failed  python2 Error: not found: python2
npm ERR! gyp verb `which` failed     at getNotFoundError (/app/node_modules/which/which.js:13:12)
npm ERR! gyp verb `which` failed     at F (/app/node_modules/which/which.js:68:19)
npm ERR! gyp verb `which` failed     at E (/app/node_modules/which/which.js:80:29)
npm ERR! gyp verb `which` failed     at /app/node_modules/which/which.js:89:16
npm ERR! gyp verb `which` failed     at /app/node_modules/isexe/index.js:42:5
npm ERR! gyp verb `which` failed     at /app/node_modules/isexe/mode.js:8:5
npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:183:21) {
npm ERR! gyp verb `which` failed   code: 'ENOENT'
npm ERR! gyp verb `which` failed }
npm ERR! gyp verb check python checking for Python executable "python" in the PATH
npm ERR! gyp verb `which` failed Error: not found: python
npm ERR! gyp verb `which` failed     at getNotFoundError (/app/node_modules/which/which.js:13:12)
npm ERR! gyp verb `which` failed     at F (/app/node_modules/which/which.js:68:19)
npm ERR! gyp verb `which` failed     at E (/app/node_modules/which/which.js:80:29)
npm ERR! gyp verb `which` failed     at /app/node_modules/which/which.js:89:16
npm ERR! gyp verb `which` failed     at /app/node_modules/isexe/index.js:42:5
npm ERR! gyp verb `which` failed     at /app/node_modules/isexe/mode.js:8:5
npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:183:21)
npm ERR! gyp verb `which` failed  python Error: not found: python
npm ERR! gyp verb `which` failed     at getNotFoundError (/app/node_modules/which/which.js:13:12)
npm ERR! gyp verb `which` failed     at F (/app/node_modules/which/which.js:68:19)
npm ERR! gyp verb `which` failed     at E (/app/node_modules/which/which.js:80:29)
npm ERR! gyp verb `which` failed     at /app/node_modules/which/which.js:89:16
npm ERR! gyp verb `which` failed     at /app/node_modules/isexe/index.js:42:5
npm ERR! gyp verb `which` failed     at /app/node_modules/isexe/mode.js:8:5
npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:183:21) {
npm ERR! gyp verb `which` failed   code: 'ENOENT'
npm ERR! gyp verb `which` failed }
npm ERR! gyp ERR! configure error 
npm ERR! gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
npm ERR! gyp ERR! stack     at PythonFinder.failNoPython (/app/node_modules/node-gyp/lib/configure.js:484:19)
npm ERR! gyp ERR! stack     at PythonFinder.<anonymous> (/app/node_modules/node-gyp/lib/configure.js:406:16)
npm ERR! gyp ERR! stack     at F (/app/node_modules/which/which.js:68:16)
npm ERR! gyp ERR! stack     at E (/app/node_modules/which/which.js:80:29)
npm ERR! gyp ERR! stack     at /app/node_modules/which/which.js:89:16
npm ERR! gyp ERR! stack     at /app/node_modules/isexe/index.js:42:5
npm ERR! gyp ERR! stack     at /app/node_modules/isexe/mode.js:8:5
npm ERR! gyp ERR! stack     at FSReqCallback.oncomplete (node:fs:183:21)
npm ERR! gyp ERR! System Linux 5.4.0-1031-azure
npm ERR! gyp ERR! command "/usr/local/bin/node" "/app/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! gyp ERR! cwd /app/node_modules/node-sass
npm ERR! gyp ERR! node -v v15.0.1
npm ERR! gyp ERR! node-gyp -v v3.8.0
npm ERR! gyp ERR! not ok 
npm ERR! Build failed with error code: 1

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-10-30T06_09_39_181Z-debug.log
The command '/bin/sh -c npm ci' returned a non-zero code: 1
Error: exit status 1
exit status 1
Usage:
  github-actions build-push [flags]

Flags:
  -h, --help   help for build-push
```

**Comment by entrotech** [comment]
For now, the workaround is to build the docker container and push it to docker hub manually as described in /docs/deployment.md.
